File: server/models/task.js
import mongoose from "mongoose";

const taskSchema=new mongoose.Schema({
    name:String,
    type:String,
    project:{
        type:mongoose.Types.ObjectId,
        ref:"Project"
    },
    createdby:{
        type:mongoose.Types.ObjectId,
        ref:"User"
    },
    affectedTo:{
        type:mongoose.Types.ObjectId,
        ref:"User"
    },
    createdAt:Date,
    deadline:Date,
    completed:Boolean,
    progress:{
        type:Number,
        min:0,
        max:100,
    },
});
const Task=mongoose.model("tasks",taskSchema);

export default Task;

File: client/team_track/lib/models/task.dart
import 'package:team_track/models/project.dart';
import 'package:team_track/models/user.dart';

class Task {
  late String name;
  late String type;
  late Project project;
  late User createdBy;
  late User affectedTo;
  late DateTime createdAt;
  late DateTime deadline;
  late bool completed;
  late double progress;

  Task({
    required this.name,
    required this.type,
    required this.project,
    required this.createdBy,
    required this.affectedTo,
    required this.createdAt,
    required this.deadline,
    required this.completed,
    required this.progress,
  });
}